src/timedate/timedated.c

```diff
@@ -916,6 +916,31 @@ static int method_set_ntp(sd_bus_message *m, void *userdata, sd_bus_error *error
         return sd_bus_reply_method_return(m, NULL);
 }
 
+static int method_list_timezones(sd_bus_message *m, void *userdata, sd_bus_error *error) {
+        sd_bus *bus;
+        Context *c = userdata;
+        int r;
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        _cleanup_strv_free_ char **zones = NULL;
+
+        assert(m);
+        assert(c);
+
+        bus = sd_bus_message_get_bus(m);
+        assert(bus);
+
+        r = get_timezones(&zones);
+        if (r < 0)
+                return sd_bus_error_setf(error, BUS_ERROR_JOB_FAILED, "Failed to read list of time zones %d", r);
+
+        r = sd_bus_message_new_method_return(m, &reply);
+        if (r < 0)
+                return sd_bus_error_set_errno(error, r);
+
+        r = sd_bus_message_append_strv(reply, zones);
+        return sd_bus_send(bus, reply, NULL);
```

Comment:
you can leave the first argument NULL, in which it is implicitly derive from the reply, and thus you don't have to query the bus yourself